app/lib/rocketchat.js

```diff
@@ -1123,6 +1123,9 @@ const RocketChat = {
 	useInviteToken(token) {
 		// RC 2.4.0
 		return this.sdk.post('useInviteToken', { token });
+	},
+	canAccessRoom(rid, uid) {
```

Comment:
We have a similar logic [here](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/lib/methods/canOpenRoom.js).
Is it the same?

Reply:
With what I've understood, no. The api call in `canAccessRoom()` returns true if the user is already in the room(being private or public) and if the user could enter in the room(being public) and this cover hidden rooms too.